### PR TITLE
Update wasmtime to (almost) lastest master

### PR DESCRIPTION
**Pull request by NikVolf**
Unfortunately, we still need a fork of wasmtime for that (because InstanceHandle we use for globals snapshot is no longer exposed). But it will be fork of only `wasmtime` crate and not others.

### CODE REVIEW

**Comment by NikVolf** [comment]
@gnunicorn 

Testing this with polkadot now

if successful, can we deploy `substrate-wasmtime 0.19` from `paritytech/wasmtime`, `update-upstream` branch?

Then, if it works as expected, we can just reset `master` to `update-upstream` branch here.

**Comment by NikVolf** [comment]
Synced polkadot fine

File: client/executor/wasmtime/Cargo.toml [hunk 1]
```diff
@@ -21,11 +21,11 @@ sp-wasm-interface = { version = "2.0.0-rc4", path = "../../../primitives/wasm-in
...
-wasmtime = { package = "substrate-wasmtime", version = "0.16.0-threadsafe.4" }
-wasmtime-runtime = { package = "substrate-wasmtime-runtime", version = "0.16.0-threadsafe.4" }
-wasmtime-environ = "0.16"
-cranelift-wasm = "0.63"
-cranelift-codegen = "0.63"
+wasmtime = { package = "substrate-wasmtime", git = "https://github.com/paritytech/wasmtime", branch = "update-upstream" }
+wasmtime-runtime = { git = "https://github.com/paritytech/wasmtime", branch = "update-upstream" }
+wasmtime-environ = { git = "https://github.com/paritytech/wasmtime", branch = "update-upstream" }
+cranelift-wasm = { git = "https://github.com/paritytech/wasmtime", branch = "update-upstream" }
+cranelift-codegen = { git = "https://github.com/paritytech/wasmtime", branch = "update-upstream" }
```
Comment (gnunicorn) [inline client/executor/wasmtime/Cargo.toml:28]: Need to put this onto a released version once tested or we can't publish this crate...

File: client/executor/wasmtime/src/instance_wrapper/globals_snapshot.rs [hunk 2]
```diff
@@ -40,7 +40,7 @@ impl GlobalsSnapshot {
 	pub fn take(instance_wrapper: &InstanceWrapper) -> Result<Self> {
 		// EVIL:
 		// Usage of an undocumented function.
-		let handle = unsafe { instance_wrapper.instance.handle().clone() };
+		let handle = instance_wrapper.instance.handle().clone().handle;
```
Comment (gnunicorn) [inline client/executor/wasmtime/src/instance_wrapper/globals_snapshot.rs:43]: yay, one `unsafe` less!

File: client/executor/wasmtime/Cargo.toml (hunk 1 above)
Comment (NikVolf) [reply client/executor/wasmtime/Cargo.toml:28]: I will change this once you publish `substrate-wasmtime` from `paritytech/wasmtime/update-upstream`

**Comment by gavofyork** [comment]
@ddorgan did this burn in ok?

**Comment by ddorgan** [comment]
This was only deployed today, will report back the status tomorrow once it's been running overnight.

**Comment by gavofyork** [comment]
@ddorgan feedback?

**Comment by ddorgan** [comment]
@gavofyork spoke to @NikVolf on what I should be looking for. Block production, basic host stats etc... are looking good.

File: client/executor/wasmtime/Cargo.toml (hunk 1 above)
Comment (gnunicorn) [reply client/executor/wasmtime/Cargo.toml:28]: is this published now? Did we merge this with github links?

Comment (NikVolf) [reply client/executor/wasmtime/Cargo.toml:28]: Not published, can you publish substrate-wasmtime 0.19 like I asked above?

I will add prepare update pr after

Comment (gnunicorn) [reply client/executor/wasmtime/Cargo.toml:28]: ah, sorry, didn't see you wanted me to do this. on it.

Comment (gnunicorn) [reply client/executor/wasmtime/Cargo.toml:28]: @NikVolf substrate-wasmtime 0.19.0 is published.

**Pull request merged**